### DISCUSSION TITLE
Add milk ratio to options.

### DISCUSSION THREAD

**Discussion by horhokokki** [thoukydides/homebridge-homeconnect#135] [Ideas]
I can make coffee with milk mix, but there is no milk ratio available to chnage? Can this be added somehow or ssomething?

**Comment by thoukydides** [comment]
If it can be supported then it already is. This plugin supports all program options that are available via the Home Connect API. None of the program options are hard-coded.

You can see what options the API supports here: https://api-docs.home-connect.com/programs-and-options?#coffee-machine

If you want additional options then you will need to convince the Home Connect team to add them:
https://developer.home-connect.com/support/contact

**Comment by horhokokki** [comment]
Weird, in IFTTT applet I can choose milk ratio, too.

**Reply by horhokokki** [reply]
And from api page.. 
ConsumerProducts.CofeeMaker.Program.Beverage.CaffeLatte
Half coffee, half warm milk; typically served in a bowl

That is wrong too, I got 40% milk now, not half. So it is not directly from the app page what we get. Luckily it is 40%. But so sad this is not done in right way at all. It would be nice to have binding to favorite .. if there is no other way to get this working right.

And what I see in api page.. there is plenty of options that are not shown in plugin.. cofee temoerature, flowrate, beam amount,

**Reply by thoukydides** [reply]
IFTTT does not use the public Home Connect API. It often has features and functions that cannot be accessed via the home Connect API. The gap has been reduced over the last few years as the API has been extended, but I understand that the IFTTT integration has recently been enhanced, so has leapt ahead again.

The API documentation does have some  errors in the detailed description of behaviour. But the fact remains that it only supports a fraction of the appliances' functionality, and this plugin cannot support anything that isn't exposed by the API. 

Like I said, you need to lobby the Home Connect team to add support. Once it has been added to the API this plugin will automatically support it without requiring any changes.
